utils/rector/src/MethodCallToFuncCallWIthObjectASFirstParameter/MethodCallToFuncCallWIthObjectAsFirstParameterRector.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the zenstruck/foundry package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Foundry\Utils\Rector\MethodCallToFuncCallWIthObjectASFirstParameter;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Name\FullyQualified;
+use PHPStan\Type\ObjectType;
+use Rector\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Rector\AbstractRector;
+use Webmozart\Assert\Assert;
+
+final class MethodCallToFuncCallWIthObjectAsFirstParameterRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    /**
+     * @var MethodCallToFuncCallWithObjectAsFirstParameter[]
+     */
+    private array $methodCallsToFuncCalls = [];
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes() : array
+    {
+        return [MethodCall::class];
+    }
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node) : ?Node
```

Comment:
I think it's this one, right?

I'll make notes as it comes to my mind:

* I'd avoid configuration and use values here instead. The fluent call has to be converted fully, to its split functoins.
* If you want to return multiple stmts = standalone lines with function calls, you need to hook into and `Expression` node. That way we're working with lines, and can return multiple lines.
* The `Expression` node must contain `$node->expr` of `MethodCall` instance, then you create functions and return them all.

First notes to kick off.